packages/slate-react/src/components/leaf.tsx

```diff
@@ -92,7 +92,7 @@ const MemoizedLeaf = React.memo(Leaf, (prev, next) => {
     next.renderLeaf === prev.renderLeaf &&
     next.text === prev.text &&
     next.leaf.text === prev.leaf.text &&
-    Text.matches(next.leaf, prev.leaf) &&
+    Text.equals(prev.leaf, next.leaf, { loose: true }) &&
```

Comment:
Great catch @gyh9457! Would it be possible to remove the previous line now (and the `loose` option) to simplify the logic even further. Instead of:

```
     next.text === prev.text &&
     next.leaf.text === prev.leaf.text &&
     Text.equals(prev.leaf, next.leaf, { loose: true }) &&
```

We'd have:

```
     next.text === prev.text &&
     Text.equals(prev.leaf, next.leaf) &&
```

Reply:
done   @ianstormtaylor